tritium-samples/src/main/java/com/palantir/tritium/samples/Sampler.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tritium.samples;
+
+import java.util.List;
+
+interface Sampler {
+    void observe(long value);
+
+    List<Sample> collect();
```

Comment:
is the intent here to be able to use something like a ring buffer to keep last-N or specific outlier samples?